File: akka-samples/java-samples/src/main/java/actors/behavior/BehaviorPrinter.java
package actors.behavior;

import akka.actor.typed.ActorSystem;
import akka.actor.typed.Behavior;
import akka.actor.typed.javadsl.AbstractBehavior;
import akka.actor.typed.javadsl.ActorContext;
import akka.actor.typed.javadsl.Behaviors;
import akka.actor.typed.javadsl.Receive;

class BehaviorPrinter extends AbstractBehavior<BehaviorPrinter.PrintMessage> {
    static final class PrintMessage {
        public final String text;

        public PrintMessage(String text) {
            this.text = text;
        }
    }

    @Override
    public Receive<PrintMessage> createReceive() {
        return newReceiveBuilder().onMessage(PrintMessage.class, (message) -> {
            System.out.println(message.text);

            return newReceiveBuilder().onMessage(PrintMessage.class, (msg) -> {
                System.out.println("Accepting no more messages!");
                return Behaviors.stopped();
            }).build();
        }).build();
    }

    public BehaviorPrinter(ActorContext<PrintMessage> context) {
        super(context);
    }

    public static Behavior<PrintMessage> create() {
        return Behaviors.setup(BehaviorPrinter::new);
    }

    public static void main(String[] args) {
        final ActorSystem<PrintMessage> system = ActorSystem.create(BehaviorPrinter.create(), "printer");
        system.tell(new PrintMessage("Hello there!"));
        system.tell(new PrintMessage("Oh Hi there!"));
    }
}

File: akka-samples/scala-samples/src/main/scala/actors/behavior/BehaviorPrinter.scala
package actors.behavior

import akka.actor.typed.scaladsl.Behaviors
import akka.actor.typed.{ActorSystem, Behavior}
import actors.behavior.Printer.PrintMessage

object BehaviorPrinter {
  def main(args: Array[String]): Unit = {
    val system = ActorSystem[PrintMessage](Printer(), "behavior-printer")
    system ! PrintMessage("Hello there!")
    system ! PrintMessage("Oh Hi there!")
  }
}

private object Printer {
  final case class PrintMessage(text: String)

  def apply(): Behavior[PrintMessage] =
    Behaviors.receive[PrintMessage] { (context, message) =>
      println(message.text)

      Behaviors.receive[PrintMessage] { (context, message) =>
        println("Accepting no more messages!")
        Behaviors.stopped
      }
    }
}